### PR TITLE
Remove NoUnexpectedThunks

### PR DESCRIPTION
**Pull request by mrBliss**
This code has been released as a separate library on Hackage:
https://hackage.haskell.org/package/nothunks

Note that we intentionally don't re-export it, so that we steer downstream repos
away from this library (i.e. `cardano-prelude`).

### CODE REVIEW

File: stack.yaml [hunk 1]
```diff
@@ -7,6 +7,7 @@ packages:
 extra-deps:
   - nonempty-containers-0.3.4.0
   - nonempty-vector-0.2.0.2
+  - nothunks-0.1.1.0
```
Comment (mrBliss) [inline stack.yaml:10]: Are we going to remove the `stack.yaml` from this repo too? What about `snapshot.yaml`?

**Review by intricate — APPROVED** [review]

**Review by edsko — APPROVED** [review]
Yay :)

**Review by nc6 — APPROVED** [review]
Great!

File: stack.yaml (hunk 1 above)
Comment (intricate) [reply stack.yaml]: I think we will, but let's do that in another PR.

**Pull request merged**